blaise_survey_manager/sample_intake/functions.py

```diff
@@ -219,32 +220,45 @@ def process_feed_forward(request):
     try:
         log.info("Loading payload from JSON request {} ".format(json.dumps(request.json)))
         payload = json.loads(request.json)
-    except Exception as ex:
-        log.error("Error loading JSON - " + str(ex))
-    try:
-        sample_header = SampleHeader.query.filter_by(survey_id=payload['survey_id'],
-                                                     field_period_id=payload['field_period_id'],
-                                                     phase=payload['phase']).first()
-
-        set_sample_header_status(sample_header, 'Uploading feed forward data')
-
-        for record in payload['feed_forward_data']:
-            case = Case.query.filter_by(survey_id=payload['survey_id'],
-                                        field_period_id=payload['field_period_id'],
-                                        serial_number=record['serial_number']).first()
-            if case:
-                log.info("Case found, setting case.fed_forward_data for record: {}".format(json.dumps(record)))
-                case.fed_forward_data = json.dumps(record)
-        db.session.commit()
-        log.info("commit carried out on session")
 
-        set_sample_header_status(sample_header, 'Uploaded')
-
-        log.info('Feed forward data upload complete')
-
-        return True
+        survey_id = payload.get('survey_id')
+        field_period_id = payload.get('field_period_id')
+        phase = payload.get('phase')
+
+        if survey_id and field_period_id and phase:
+            sample_header = SampleHeader.query.filter_by(survey_id=payload['survey_id'],
+                                                         field_period_id=payload['field_period_id'],
+                                                         phase=payload['phase']).first()
+            if sample_header:
+                set_sample_header_status(sample_header, 'Uploading feed forward data')
+
+                for record in payload['feed_forward_data']:
+                    case = Case.query.filter_by(survey_id=payload['survey_id'],
+                                                field_period_id=payload['field_period_id'],
+                                                serial_number=record['serial_number']).first()
```

Comment:
As you already have pulled the items from the payload, is it worth using them again in the query, I think this would be more efficient and cleaner. 
```suggestion
        survey_id = payload.get('survey_id')
        field_period_id = payload.get('field_period_id')
        phase = payload.get('phase')

        if survey_id and field_period_id and phase:
            sample_header = SampleHeader.query.filter_by(survey_id=survey_id,
                                                         field_period_id=field_period_id,
                                                         phase=payload['phase']).first()
            if sample_header:
                set_sample_header_status(sample_header, 'Uploading feed forward data')

                for record in payload['feed_forward_data']:
                    case = Case.query.filter_by(survey_id=survey_id,
                                                field_period_id=field_period_id,
                                                serial_number=record['serial_number']).first()
```

Reply:
Yes that's a good shout - makes sense